logos/src/logos/dbutils/dbmanager.py

```
@@ -5118,13 +5283,96 @@ def get_team_budget_usage(self, team_id: int, month_start: str) -> int:
+                      AND valid_from <= le.timestamp_request
+                    ORDER BY (model_id = le.model_id) DESC NULLS LAST,
+                             (provider_id = le.provider_id) DESC NULLS LAST,
+                             valid_from DESC
+                    LIMIT 1
+                ) tp ON true
+                WHERE le.timestamp_request >= :start
+                  AND le.timestamp_request < :end
+                  AND le.api_key_id IS NOT NULL
+                GROUP BY t.id, t.name, DATE_TRUNC('{interval}', le.timestamp_request)
+                ORDER BY bucket_ts, t.name
+                """
```

Ruff (0.15.15)

[error] Possible SQL injection vector through string-based query construction

(S608)